scipy/optimize/tests/test__basinhopping.py

```diff
@@ -171,6 +171,19 @@ def test_njev(self):
         assert_(res.nfev > 0)
         assert_equal(res.nfev, res.njev)
 
+    def test_jac(self):
+        # test jacobian returned
+        i = 1
+        minimizer_kwargs = self.kwargs.copy()
+        # BFGS returns a Jacobian
+        minimizer_kwargs["method"] = "BFGS"
+
+        res = basinhopping(func2d, self.x0[i],
+                           minimizer_kwargs=minimizer_kwargs, niter=self.niter,
+                           disp=self.disp)
+
+        assert_(hasattr(res, "jac"))
+
```

Comment:
you might want to check somewhere that it returns the correct jacobian, as opposed to any jacobian.